### PR TITLE
Allow Support Users to change Course to previous Recruitment Cycles

### PR DESCRIPTION
**Pull request by dcyoung-dev**
## Context

Developers are often [requested](https://trello.com/c/If5FF45J) to change a Course to one in a previous cycle. 
This functionality is also what is required for Support Users to confirm deferrals in previous cycles when they want to change course. 

## Changes proposed in this pull request

Minor UI change to the change-course-choice UI to add a "Recruitment cycle year" input. This input defaults to the current cycle year. 

| before | after |
|---|---|
| image | image |

## Guidance to review

- Login to Support console on Review App
- Select an Application Form
- Scroll to Application Choice and click "Change Course"
- Enter course details for a Course in a previous cycle. 
- Submit form
- See the change

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [ ] Make sure all information from the Trello card is in here
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Add PR link to Trello card

### CODE REVIEW

**Comment by dcyoung-dev** [comment]
[The ability to confirm deferrals across courses via Apply Support Console](https://trello.com/c/7vJc3JLA/233-the-ability-to-confirm-deferrals-across-courses-via-apply-support-console)

**Review by CatalinVoineag — APPROVED** [review]
Tested it on review, works as expected

File: app/views/support_interface/application_forms/courses/edit.html.erb [hunk 1]
```diff
@@ -32,6 +32,7 @@
         <%= f.govuk_radio_button :study_mode, :part_time, label: { text: 'Part time' } %>
       <% end %>
       <%= f.govuk_text_field :site_code, label: { text: 'Site code', size: 's' }, width: 5 %>
+      <%= f.govuk_text_field :recruitment_cycle_year, label: { text: 'Recruitment cycle year', size: 's' }, width: 5 %>
```
Comment (avinhurry) [inline app/views/support_interface/application_forms/courses/edit.html.erb:35]: Would the main use case for this be for changing it in the current cycle and the previous cycles? If so, how do you feel about using a couple of radios, similar to study mode? 

It feels a bit wrong to leave it so open with a text field when we're mainly going to be entering 2025 and 2024 in there.

Comment (dcyoung-dev) [reply app/views/support_interface/application_forms/courses/edit.html.erb:35]: Good shout - I'll ask the Support team to see which they'd prefer 👌

Comment (dcyoung-dev) [reply app/views/support_interface/application_forms/courses/edit.html.erb:35]: Also worth noting that myself and @PeteWilliams caught up to discuss the needs of this page as well as the current design. This is something we'll look at in future 👌

**Pull request merged**